components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -77,6 +78,22 @@ export const onConnectHardwareWallet = (opts: HardwareWalletConnectOpts): Promis
           reject(result.error)
         })
         .catch(reject)
+    } else if (keyring instanceof SolanaLedgerKeyring && opts.network) {
+      keyring.getAccounts(opts.startIndex, opts.stopIndex)
+        .then(async (result: GetAccountsHardwareOperationResult) => {
+          if (result.payload) {
+            const { braveWalletService } = getAPIProxy()
+            const addressesEncoded = await braveWalletService.base58Encode(
+              result.payload.map((hardwareAccount) => [...(hardwareAccount.addressBytes || [])])
+            )
+            for (let i = 0; i < result.payload.length; i++) {
+              result.payload[i].address = addressesEncoded.addresses[i]
+            }
+            return resolve(result.payload)
```

Comment:
Unlike the other cases in this conditional, if the keyring is a SolanaLedgerKeyring then the addresses retrieved from it are actually Buffers (not strings).  As a result we can't immediately display them to the user before they add accounts.  We must first base58 encode the buffer data.  Instead of adding another js library for this purpose, I instead exposed a mojo API, which is used above.